### PR TITLE
Change signMessage to local sign with web3j

### PR DESCRIPTION
**Pull request by grimadas**
- Add specific format needed for ethereum contracts

### CODE REVIEW

File: src/main/kotlin/sidechain/eth/util/Crypto.kt [hunk 1]
```diff
@@ -4,26 +4,31 @@ import config.EthereumConfig
...
  * @return signed data
  */
 fun signUserData(ethereumConfig: EthereumConfig, ethereumPasswords: EthereumPasswords, toSign: String): String {
-    // TODO luckychess 26.06.2018 D3-100 find a way to produce correct signatures locally
-    val deployHelper = DeployHelper(ethereumConfig, ethereumPasswords)
 
-    val admin = Admin.build(HttpService(ethereumConfig.url))
-    admin.personalUnlockAccount(deployHelper.credentials.address, ethereumPasswords.credentialsPassword).send()
-    // TODO: signalize about error if unlock fails
+    val deployHelper = DeployHelper(ethereumConfig, ethereumPasswords)
```
Comment (dolgopolovwork) [inline src/main/kotlin/sidechain/eth/util/Crypto.kt]: can we move deployHeper to the top level?

Comment (grimadas) [reply src/main/kotlin/sidechain/eth/util/Crypto.kt]: Pass with params?

Comment (grimadas) [reply src/main/kotlin/sidechain/eth/util/Crypto.kt]: Actually, we can even remove it now. Good point

File: src/main/kotlin/sidechain/eth/util/Crypto.kt [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
+    val dat = Numeric.hexStringToByteArray(toSign)
+    // Add ethereum signature format
+    val to_sign = ("\u0019Ethereum Signed Message:\n" + (dat.size)).toByteArray()
+    val data_sign = to_sign.plus(dat)
+    val signature = Sign.signMessage(data_sign, ecKeyPair)
+    // Combine in the signature
+    var res = Numeric.toHexString(signature.r)
+    res = res.plus(Numeric.toHexString(signature.s).substring(2))
+    //  The v is always either 27, or 28 - need to convert to 00 or 01
+    res = res.plus("0" + Integer.toString(signature.v.toInt() - 27))
```
Comment (dolgopolovwork) [inline src/main/kotlin/sidechain/eth/util/Crypto.kt]: what is 27? seems like magic constant. can we create named constant for this?

Comment (grimadas) [reply src/main/kotlin/sidechain/eth/util/Crypto.kt]: It is Ethereum magic constant, we gonna is only here, so maybe there is no need for naming

**Pull request merged**